core/src/main/java/com/sequenceiq/cloudbreak/converter/StackToTemplatePreparationObjectConverter.java

```diff
@@ -392,6 +394,14 @@ private boolean isCloudStorageConfigured(Stack source) {
 
     private GeneralClusterConfigs calculateGeneralClusterConfigs(Stack source, Cluster cluster) {
         GeneralClusterConfigs generalClusterConfigs = generalClusterConfigsProvider.generalClusterConfigs(source, cluster);
+        boolean allInstanceGroupsHaveMultiAz = source.getInstanceGroups().stream().allMatch(ig -> {
+            boolean instanceGroupHasMultiAz = ((List<String>) ig.getInstanceGroupNetwork()
+                    .getAttributes()
+                    .getMap()
+                    .getOrDefault(NetworkConstants.SUBNET_IDS, new ArrayList<>())).size() > 1;
```

Comment:
just by reading the relevant code part, it does not imply that if the SUBNET_IDS list size is greater than 1 then it would necessarily mean that those subnets are in different AZs

Reply:
What would be the correct way to detect if they are in different AZs?

Reply:
cc @biharitomi @topolyai5

Reply:
As I see `instanceGroup.getAvailabilityZones()` could be better (just to answer myself quickly :) but please correct me if not.